lib/nest_thermostat/nest.rb

```diff
@@ -42,96 +42,92 @@ def initialize(config = {})
     end
 
     def status
-      request = HTTParty.get("#{self.transport_url}/v2/mobile/user.#{self.user_id}", headers: self.headers) rescue nil
-      result = JSON.parse(request.body) rescue nil
+      request = HTTParty.get("#{self.transport_url}/v2/mobile/user.#{self.user_id}", headers: self.headers)
+      result = JSON.parse(request.body)
 
-      self.structure_id = result['user'][user_id]['structures'][0].split('.')[1]
-      self.device_id    = result['structure'][structure_id]['devices'][0].split('.')[1]
+      structures = result['user'][user_id]['structures']
+      if structures.any?
+        self.structure_id = structures.first.split('.')[1]
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.